src/Lean/Util/Trace.lean

```diff
@@ -193,7 +193,9 @@ def withTraceNode [MonadExcept ε m] [MonadLiftT BaseIO m] (cls : Name) (msg : E
     modifyTraces (oldTraces ++ ·)
     return (← MonadExcept.ofExcept res)
   let ref ← getRef
-  let mut m ← msg res
+  let mut m := match ← observing <| msg res with
+    | .ok m => m
+    | .error _ => m!"<exception thrown while producing trace node message>"
```

Comment:
```suggestion
  let mut m ← try msg res catch _ => m!"<exception thrown while producing trace node message>"
```

Do you see an easy way to make the `<exception ...>` part interactive?  (I've often wished that I could hover over the 💣 emoji and see what went wrong.)

Reply:
What kind of interaction do you have in mind? You can write your message action like `fun | .error e => m!"💣 {e.toMessageData}" | ...`, and this will be interactive if the error is. Do you mean something else?

Reply:
Oh, I meant interactive in the sense that it only shows the emoji by default and you need to hover to see the exception message.

But maybe we should just always print the exception.

Reply:
Ah, we could probably do that with #2064 :)